include/api/wasmedge/wasmedge.h

```diff
@@ -138,6 +144,65 @@ typedef struct WasmEdge_Async WasmEdge_Async;
 /// Opaque struct of WasmEdge VM.
 typedef struct WasmEdge_VMContext WasmEdge_VMContext;
 
+/// Type of option value.
+typedef enum WasmEdge_ProgramOptionType {
+  /// No option value.
+  WasmEdge_ProgramOptionType_None,
+  /// Boolean value.
+  WasmEdge_ProgramOptionType_Toggle,
+  WasmEdge_ProgramOptionType_Int8,
+  WasmEdge_ProgramOptionType_Int16,
+  WasmEdge_ProgramOptionType_Int32,
+  WasmEdge_ProgramOptionType_Int64,
+  WasmEdge_ProgramOptionType_UInt8,
+  WasmEdge_ProgramOptionType_UInt16,
+  WasmEdge_ProgramOptionType_UInt32,
+  WasmEdge_ProgramOptionType_UInt64,
+  WasmEdge_ProgramOptionType_Float,
+  WasmEdge_ProgramOptionType_Double,
+  /// WasmEdge_String.
+  WasmEdge_ProgramOptionType_String,
+} WasmEdge_ProgramOptionType;
+
+/// Program option for plugins.
+typedef struct WasmEdge_ProgramOption {
+  const char *Name;
+  const char *Description;
+  WasmEdge_ProgramOptionType Type;
+  void *Storage;
+  const void *DefaultValue;
+} WasmEdge_ProgramOption;
+
+/// Module descriptior for plugins.
+typedef struct WasmEdge_ModuleDescriptor {
+  const char *Name;
+  const char *Description;
+  WasmEdge_ModuleInstanceContext *(*Create)(
+      const struct WasmEdge_ModuleDescriptor *);
+} WasmEdge_ModuleDescriptor;
+
+/// Version data for plugins.
+typedef struct WasmEdge_PluginVersionData {
+  uint32_t Major;
+  uint32_t Minor;
+  uint32_t Patch;
+  uint32_t Build;
+} WasmEdge_PluginVersionData;
+
+/// Plugin descriptior for plugins.
+typedef struct WasmEdge_PluginDescriptor {
+  const char *Name;
+  const char *Description;
+  uint32_t APIVersion;
+  WasmEdge_PluginVersionData Version;
+  uint32_t ModuleCount;
+  uint32_t ProgramOptionCount;
+  WasmEdge_ModuleDescriptor *ModuleDescriptions;
+  WasmEdge_ProgramOption *ProgramOptions;
+} WasmEdge_PluginDescriptor;
+
+#define WasmEdge_Plugin_CurrentAPIVersion UINT32_C(2)
```

Comment:
Should we use cmake to define this number? I found that this version will be used in both `wasmedge.h` and `plugin/plugin.h`.

Reply:
Moved.